### PR TITLE
Uni 165: Scenarios for boat compatible

### PR DESCRIPTION
**Pull request by abanik519**
- Created a scenario that selects a random boat from the "Boat (compatible with)" facet and selects a random product
- Updated 'CoveoSelectFacetValueByText.js' to account for collapsed facets
- Added a pagination probability into the input of 'CoveoClickResultAtRandom'

### CODE REVIEW

File: tests/Barca.Engineering.boat_compatible_products.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const ExtractedQueries = require("../input/data/barca-engineering-data/boats-list.json");
```
Comment (jdevost) [inline tests/Barca.Engineering.boat_compatible_products.spec.js:1]: We shouldn't need a pre-filled list of values. 

I am expecting something more like: 
```javascript
SelectRandomValueInFacet(page, 'eng_boat'); // using facetid, field or title
```

So we could then reuse the same command for other facets like Power Output, Battery Type or Memory.
We wouldn't want to extract values for each of these facets.

File: tests/RandomVisit.Barca.Engineering.spec.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ test.describe('Scenario for Barca Engineering', () => {
   const performSearchAndSelectCategory = async function (page, query, facetSelection) {
     await CoveoPause(page, 1000);
     await CoveoSearch(page, query);
+    await page.pause();
     await CoveoSelectFacetValueByText(page, query, facetSelection, 'category', 'barca-eng');
```
Comment (jdevost) [inline tests/RandomVisit.Barca.Engineering.spec.js:32]: While at it, can you replace the use of parameter `demoSite` by using the `process.env.SELECTORS` ?

Now that we have a global variable, we don't need to pass around a parameter like `demoSite` in functions. 
So `demoSise === "barca-eng` can be replaced by `process.env.SELECTORS === 'BarcaEngineering' `

**Review by jdevost — CHANGES_REQUESTED** [review]

File: tests/RandomVisit.Barca.Engineering.spec.js [hunk 3]
```diff
@@ -28,7 +28,8 @@ test.describe('Scenario for Barca Engineering', () => {
   const performSearchAndSelectCategory = async function (page, query, facetSelection) {
     await CoveoPause(page, 1000);
     await CoveoSearch(page, query);
-    await CoveoSelectFacetValueByText(page, query, facetSelection, 'category', 'barca-eng');
+    await page.pause();
```
Comment (pkuttappan) [inline tests/RandomVisit.Barca.Engineering.spec.js:31]: This should be removed or commented out

**Review by pkuttappan — CHANGES_REQUESTED** [review]

File: input/data/barca-engineering-data/boats-list.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "Mariner",
+    "Mercruiser",
+    "Mercury",
+    "Nissan",
+    "OMC",
+    "Sears",
+    "Suzuki",
+    "Tohatsu",
+    "Yamaha"
+  ]
```
[the remark this answers is not in the document]
Comment (abanik519) [reply input/data/barca-engineering-data/boats-list.json]: No I forgot to delete it, good catch!

File: commands/CoveoSelectFacetValueByText.js [hunk 5]
```diff
@@ -150,7 +187,7 @@ async function CoveoSelectFacetValueByText(
   }
   else {
     const breadcrumbPath = Array.isArray(facetValue) ? facetValue.join('|') : null;
-    res = await selectFacetValueRecursive(page, expect, facetValue, facetName, breadcrumbPath, demoSite);
+    res = await selectFacetValueRecursive(page, facetValue, facetName, breadcrumbPath, demoSite);
```
Comment (jdevost) [inline commands/CoveoSelectFacetValueByText.js:190]: I am getting a failure on 'demoSite' not being defined. 

`demoSite` need to be clean up in this file.

**Review by jdevost — APPROVED** [review]

**Pull request merged**